src/composables/transactionList.ts

```diff
@@ -0,0 +1,338 @@
+import { useStore } from 'vuex';
+import { orderBy, uniqBy } from 'lodash-es';
+import {
+  computed,
+  onMounted,
+  ref,
+  watch,
+} from 'vue';
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+
+import type {
+  ITransaction,
+  ITransactionsState,
+  ITransactionStateList,
+} from '../types';
+import { useMiddleware } from './middleware';
+import {
+  AEX9_TRANSFER_EVENT,
+  fetchJson, getLocalStorageItem, setLocalStorageItem,
+  TX_DIRECTION,
+} from '../popup/utils';
+import { useAccounts } from './accounts';
+import JsonBig from '../lib/json-big';
+import { useAeSdk } from './aeSdk';
+
+const LOCAL_STORAGE_KEY = 'transactions_state';
+
+const isInitalized = ref(false);
+
+const transactions = ref<ITransactionStateList>({});
+
+function generateEmptyTransactionState(): ITransactionsState {
+  return {
+    loaded: [],
+    nextPageUrl: '',
+    pending: {},
+    tipWithdrawnTransactions: [],
+  };
+}
+function getTransactionStateByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value[address];
+}
+
+function getOrInitTransactionStateForAccount(address: Encoded.AccountAddress): ITransactionsState {
+  const accountTransactionState = getTransactionStateByAccountAddress(address);
+
+  if (accountTransactionState) {
+    return accountTransactionState;
+  }
+
+  transactions.value[address] = generateEmptyTransactionState();
+
+  return transactions.value[address];
+}
+
+function getTransactionByHash(address: Encoded.AccountAddress, hash: string) {
+  return transactions.value[address]?.loaded.find((transaction) => transaction.hash === hash);
+}
+
+function updateAccountTransactions(address: Encoded.AccountAddress, transaction: ITransaction) {
+  const transactionIndex = transactions.value[address]?.loaded?.findIndex(
+    (tr) => tr.hash === transaction.hash,
+  );
+
+  if (transactionIndex >= 0) {
+    transactions.value[address].loaded[transactionIndex] = transaction;
+  } else {
+    transactions.value[address].loaded.push(transaction);
+  }
+}
+
+const setTransactionsNextPage = (address: Encoded.AccountAddress, url: string) => {
+  transactions.value[address].nextPageUrl = url;
+};
+
+export function useTransactionList() {
+  const store = useStore();
+  const { nodeNetworkId, getAeSdk } = useAeSdk({ store });
+  const { activeAccount, isLoggedIn } = useAccounts({ store });
+
+  const allTransactionsForAccount = computed(() => {
+    if (!isLoggedIn) {
+      return [];
+    }
+    const { pending, loaded } = getOrInitTransactionStateForAccount(activeAccount.value.address);
+
+    return [...loaded, ...(pending[nodeNetworkId.value!] || [])];
+  });
+
+  function setPendingTransactionSentByHash(address: Encoded.AccountAddress, hash: string) {
+    getOrInitTransactionStateForAccount(address);
```

Comment:
I think this line can be removed. Just do:
```
const accountTransactionsState = getAccountTransactionsState(address);
```
And then reuse this instead of repeating `transactions.value[address]`.